FindSpicy.cmake

```diff
@@ -40,14 +40,21 @@ macro (configure)
             message(STATUS "'${SPICY_CONFIG}' does not exist")
         endif ()
     else ()
+        # Attempt to find `spicy-config` only with configured paths.
         find_program(
             spicy_config spicy-config
             HINTS ${SPICY_ROOT_DIR}/bin
                   ${SPICY_ROOT_DIR}/build/bin
                   $ENV{SPICY_ROOT_DIR}/bin
                   $ENV{SPICY_ROOT_DIR}/build/bin
                   # Try build directory of Spicy distribution we may be part of.
-                  ${PROJECT_SOURCE_DIR}/../../build/bin)
+                  ${PROJECT_SOURCE_DIR}/../../build/bin
+            NO_DEFAULT_PATH)
+
+        # If `spicy-config` was not found in configured paths try to find it in builtin search paths.
+        if (NOT spicy_config)
```

Comment:
From the docs I think technically you do not need the `if (NOT` as it will not try to find something if spicy_config is set.

Reply:
Interesting, I didn't know that. I got rid of the conditional, but added a comment so the code doesn't become shorter, but possibly more idiomatic.